### PR TITLE
Jringle/restructure

### PR DESCRIPTION
**Pull request by ringlej**
This PR supercedes https://github.com/ScenicFramework/scenic_driver_local/pull/26
I've broken up the PR into several commits to help facilitate code reviewing.

The 1st 2 commits:  7ad4b4e7ddbfdf552b6b702688098c4615c6fa6c and 96a7c02acda0f31b0909f559443aa192d658b8ff have no functional changes. They simply move source files and cleanup the formatting of the c code.

The rest of the commits are themed around addressing a part.

### CODE REVIEW

File: Makefile
Comment (crertel) [inline Makefile:1]: I like this chunking up, good thinking!

File: c_src/device/drm.c
Comment (crertel) [inline c_src/device/drm.c:544]: Similar to some other deadcode I've seen...perhaps this could be behind a `#ifdef` instead? If not, we should delete it.

File: c_src/device/drm.c
Comment (crertel) [inline c_src/device/drm.c:805]: This routine really wants to live in like a `gl_helpers.c` file.

File: c_src/device/bcm.c
Comment (crertel) [inline c_src/device/bcm.c:317]: This routine really wants to live in like a `gl_helpers.c` file.

File: c_src/device/glfw.c
Comment (crertel) [inline c_src/device/glfw.c:395]: This routine really wants to live in like a `gl_helpers.c` file.

File: c_src/image/image.h
Comment (crertel) [inline c_src/image/image.h:28]: Minor preference for `ALL_CAPS` or `kEnumValue`, but not worth holding up a PR over. :)

File: c_src/image/ops/image_ops.c
Comment (crertel) [inline c_src/image/ops/image_ops.c:1]: What's the purpose of all the `__attribute__((weak))` weak symbol decls?

Just providing something to get overridden when linked with non-stub backend code?

File: c_src/scenic/comms.c
Comment (crertel) [inline c_src/scenic/comms.c:299]: Deadcode, possibly should be `#ifdef` code.

File: c_src/scenic/ops/script_ops.h
Comment (crertel) [inline c_src/scenic/ops/script_ops.h:33]: We should probably either zorch these or add an explanation for why they're still here.

**Review by crertel — CHANGES_REQUESTED** [review]
Really great work! A handful of little polish points, but I really like the direction you took it.

I also think, if I'm tracking correctly, the weak symbol trick for default behaviors is cool.

One thing I do want to flag is that the `log_*` family has, IIRC, an internal buffer size of like 300 or something. So, we want to keep an eye on that.

**Comment by ringlej** [comment]
> Really great work! A handful of little polish points, but I really like the direction you took it.
> 
> I also think, if I'm tracking correctly, the weak symbol trick for default behaviors is cool.
> 

Thanks for reviewing. I’ll get around to updating the PR to address your comments on Monday. 

> One thing I do want to flag is that the `log_*` family has, IIRC, an internal buffer size of like 300 or something. So, we want to keep an eye on that.

I’m not sure what your meaning. All of the `log_*()` functions funnel to the [`logv()` function](https://github.com/gridpoint-com/scenic_driver_local/blob/4222f3bdd54b29a9e603e147be44b2234acf1d7c/c_src/scenic/comms.c#L86) which calls [`vasprintf()`](https://linux.die.net/man/3/vasprintf) which allocates a buffer on the heap large enough to hold the output. The buffer is free’d at the end of `logv()`

**Comment by crertel** [comment]
> I’m not sure what your meaning. All of the `log_*()` functions funnel to the [`logv()` function](https://github.com/gridpoint-com/scenic_driver_local/blob/4222f3bdd54b29a9e603e147be44b2234acf1d7c/c_src/scenic/comms.c#L86) which calls [`vasprintf()`](https://linux.die.net/man/3/vasprintf) which allocates a buffer on the heap large enough to hold the output. The buffer is free’d at the end of `logv()`

You're right! I got mixed up thinking about [this one](https://github.com/gridpoint-com/scenic_driver_local/blob/4222f3bdd54b29a9e603e147be44b2234acf1d7c/c_src/scenic/script.c#L149).

**Comment by ringlej** [comment]
> You're right! I got mixed up thinking about [this one](https://github.com/gridpoint-com/scenic_driver_local/blob/4222f3bdd54b29a9e603e147be44b2234acf1d7c/c_src/scenic/script.c#L149).

Ah. I'll get rid of `put_msg_i()` as well. All calls to it should have been replaced with a `log_*()` function call

File: c_src/device/drm.c
Comment (ringlej) [reply c_src/device/drm.c]: removed

File: c_src/device/drm.c
Comment (ringlej) [reply c_src/device/drm.c]: Moved to `c_src/device/gl_helpers.c`

File: c_src/device/bcm.c
Comment (ringlej) [reply c_src/device/bcm.c]: Moved to `c_src/device/gl_helpers.c`

File: c_src/device/glfw.c
Comment (ringlej) [reply c_src/device/glfw.c]: Moved to `c_src/device/gl_helpers.c`

File: c_src/scenic/ops/script_ops.h
Comment (ringlej) [reply c_src/scenic/ops/script_ops.h]: Removed

File: c_src/image/image.h
Comment (ringlej) [reply c_src/image/image.h]: This enum and others that I added I changed to be `ALL_CAPS` for the enum values

File: c_src/scenic/comms.c
Comment (ringlej) [reply c_src/scenic/comms.c]: Removed

File: c_src/image/ops/image_ops.c
Comment (ringlej) [reply c_src/image/ops/image_ops.c:1]: It is there just to provide a stub as I develop cairo. I may remove the weak attribute for these

**Comment by ringlej** [comment]
> > You're right! I got mixed up thinking about [this one](https://github.com/gridpoint-com/scenic_driver_local/blob/4222f3bdd54b29a9e603e147be44b2234acf1d7c/c_src/scenic/script.c#L149).
> 
> Ah. I'll get rid of `put_msg_i()` as well. All calls to it should have been replaced with a `log_*()` function call in [b429f69](https://github.com/ScenicFramework/scenic_driver_local/commit/b429f697cc4ee4ee39873956ea4742e869db1cb3)

Remove `put_msg_i()`. It is no longer used

File: c_src/image/ops/image_ops.c
Comment (ringlej) [reply c_src/image/ops/image_ops.c:1]: I removed the weak stub functions that were in image_ops.c and font_ops.c. That was the only thing in those files, so those files were removed

**Review by crertel — APPROVED** [review]
LGTM. Maybe @boydm might also want to take a look?

**Comment by crertel** [comment]
(note that I still haven't run this locally; probably should do that before merging :P )

**Comment by ringlej** [comment]
FWIW, I've been running it with glfw on laptop with no issues. But my testing of this is only for a small subset of script operations

**Comment by crertel** [comment]
I'm going in and merging this so we can proceed further.

**Pull request merged**
